fabric_dialer.go

```diff
@@ -43,29 +43,48 @@ func (f *Fabric) DialContext(ctx context.Context, as string) error {
 		}
 
 		newAddr := newConn.GetAddress()
-
 		lgr.Info("Dial complete, negotiating",
 			zap.String("address", newAddr.String()),
 			zap.String("address.Remaining", newAddr.RemainingString()),
 		)
+		return newCtx, newConn, nil
```

Comment:
Return the first connection only? What if we have multiple connections from multiple transports??